### ISSUE TITLE
after stage0 redesign, `x check` defaults to building the compiler

### ISSUE THREAD

**Issue by jyn514** [rust-lang/rust#141902]
### Summary

since https://github.com/rust-lang/rust/pull/119899, `x check` has a global default of `--stage 1`. that makes builds for the default and compiler profiles much slower, because they are building the compiler before checking std.

we should do the following things:
- `profile = "compiler"` should set `check-stage = 0` (this should be an easy change in `defaults/compiler.toml`)
- maybe the global default should also do that?
- `x check --stage 0 std` should give a warning if it doesn't already

note that after my suggested change, `x check` without arguments will no longer check the standard library for `compiler` profile. that seems ~fine.

reported in [#t-infra/bootstrap > Stage 0 std redesign support thread @ 💬](https://rust-lang.zulipchat.com/#narrow/channel/326414-t-infra.2Fbootstrap/topic/Stage.200.20std.20redesign.20support.20thread/near/521725744).

### Command used

```sh
./x.py check
```

### Expected behaviour

`x check` is a ~fast operation, like before.

### Actual behaviour

`x check` takes about twice as long (because it defaults to `--stage 1`, which builds the compiler)

### Bootstrap configuration (bootstrap.toml)

`cp config.example.toml config.toml`

### HEAD

https://github.com/rust-lang/rust/commit/52882f6522ae9f34f1d574b2efabc4b18e691ae0

### Additional context

<details><summary>Build Log</summary>
<p>

```txt
<log>
```

</p>
</details>
